### PR TITLE
cgroups: fix a leak when using cgroupfs

### PR DESCRIPTION
**Pull request by giuseppe**
be sure to load all the existing handlers, so that they can also be freed in addition to the handlers we treat differently.
    
Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: pkg/cgroups/blkio.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func (c *blkioHandler) Create(ctr *CgroupControl) (bool, error) {
 
 // Destroy the cgroup
 func (c *blkioHandler) Destroy(ctr *CgroupControl) error {
-	return os.Remove(ctr.getCgroupv1Path(Blkio))
+	return rmDirRecursively(ctr.getCgroupv1Path(Blkio))
```
Comment (rhatdan) [inline pkg/cgroups/blkio.go]: Why can't you use
os.RemoveAll(ctr.getCgroupv1Path(Blkio))

**Comment by vrothberg** [comment]
High five :clap: ... this PR in combination with my patch for CRI-O solves the issue as well

File: pkg/cgroups/blkio.go (hunk 1 above)
Comment (giuseppe) [reply pkg/cgroups/blkio.go]: `os.RemoveAll()` would not work with cgroups.  It will try to remove all files, while we can remove just  the directories.

**Comment by baude** [comment]
@mheon, remove the hold if you are happy

**Comment by mheon** [comment]
Code looks fine, but can we get some comments about why we need a custom
remove function? I don't want someone removing it thinking it's just the
same as os.RemoveAll.

**Comment by giuseppe** [comment]
> Code looks fine, but can we get some comments about why we need a custom remove function? I don't want someone removing it thinking it's just the same as os.RemoveAll.

added a comment and repushed

**Pull request merged**